pkg/machines/libvirt-dbus.js

```diff
@@ -1393,6 +1393,18 @@ export function storagePoolUndefine(connectionName, objPath) {
     return call(connectionName, objPath, 'org.libvirt.StoragePool', 'Undefine', [], TIMEOUT);
 }
 
+export function storageVolumeCreate(connectionName, poolName, volName, size, format) {
+    const volXmlDesc = getVolumeXML(volName, size, format);
+
+    return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'StoragePoolLookupByName', [poolName], TIMEOUT)
+            .then((path) => {
+                return call(connectionName, path[0], 'org.libvirt.StoragePool', 'StorageVolCreateXML', [volXmlDesc, 0], TIMEOUT)
+                        .then(() => {
```

Comment:
Please avoid nesting promises. You can chain them, meaning that the refresh can go outside this promise block.

Reply:
Fixed according to your comments.

Reply:
Oh wait, nested promise here is in place, since it keeps scope of $path variable.

Reply:
Well you can store it in a variable and use it in the chained promise.

Reply:
Is it really nicer looking solution than just nesting that promise? I agree uneccessary nesting is bad, but nesting is not a bad thing if there is a reason for that (like keeping scope).

Reply:
There are more drawbacks than plain style preference. If the nested promise will fail the error will not get propagated to the end of the chain, resulting in unhandled error.
If you make sure there are no unhandled errors I don't have any objection.

Reply:
Error won't reach end of the chain but it will be returned. In nested promises, any error which happens along the chain will "stop the chain" and return the error. So If I do error handling in the place where I call the function from, then it fill be handled.

Reply:
If force pushed the change and tested that failure will get propagated even in nested chain.

Reply:
Not really, make Refresh fail and you will never see the error in the UI.
Try it out, just replace Refresh method name with something that doesn't exist.

Reply:
Eeh I forgot to put return in front of resfresh method call. Repushed, and with that in place, it propagates for me.
(not exactly changing refresh method name, because that would result in not loading the pool list in UI and then I wouldnt have a way to test it. But If you change path[0] parameter to path[1] of storagePoolRefresh(), then error from storagePoolRefresh will get propageted.